types/stylelint/index.d.ts

```diff
@@ -323,7 +323,7 @@ declare module 'stylelint' {
 		source?: string;
 		deprecations: {
 			text: string;
-			reference: string;
+			reference: string | undefined;
```

Comment:
[note] for more accuracte type.

Reply:
[question]

Should it be optional instead?
e.g. `reference?: string;`

i.e. reference must be set when you are explicit
e.g.
```
				deprecations: [
					{
						text: 'Deprecated foo',
						reference: undefined,
					},
				],
```

Reply:
Good point. Fixed via d6108a993.